batch/executor.go

```diff
@@ -48,21 +48,21 @@ func (b *batch[T, R]) add(task T, future *futures.Future[R]) int {
 	return len(b.tasks)
 }
 
-// BatchExecutor batches values of type T submited via multiple producers and invokes the provided run function
+// Executor batches values of type T submitted via multiple producers and invokes the provided run function
 // when a batch flushes either due to size or a timeout.  Results of type R are returned to the caller of Submit
 // when the batch finishes execution.
-// A BatchExecutor must be created by calling New
-type BatchExecutor[T any, R any] struct {
+// A Executor must be created by calling New
+type Executor[T any, R any] struct {
 	maxSize   int
 	maxLinger time.Duration
 	taskChan  chan tsk.TaskFuture[T, R]
 	run       RunBatchFunction[T, R]
 }
 
-// New creates a new BatchExecutor with the specified options that invokes the provided run function
+// New creates a new Executor with the specified options that invokes the provided run function
 // when a batch of items flushes due to either size or time.
-func New[T any, R any](opts Opts, run RunBatchFunction[T, R]) *BatchExecutor[T, R] {
-	be := &BatchExecutor[T, R]{
+func New[T any, R any](opts Opts, run RunBatchFunction[T, R]) *Executor[T, R] {
+	be := &Executor[T, R]{
```

Comment:
Left the variable and receiver names as `be`. It seemed ok to me but I can also rename them to `e` or something else.